cmd/podman/replay_kube.go

```diff
@@ -0,0 +1,247 @@
+	// TODO for now we just used the default kernel namespaces; we need to add/subtract this from yaml
+
+	nsOptions, err := shared.GetNamespaceOptions(strings.Split(DefaultKernelNamespaces, ","))
+	if err != nil {
+		return err
+	}
+	podOptions = append(podOptions, nsOptions...)
+	podPorts := getPodPorts(podYAML.Spec.Containers)
+	podOptions = append(podOptions, libpod.WithInfraContainerPorts(podPorts))
+
+	// Create the Pod
+	pod, err := runtime.NewPod(ctx, podOptions...)
+	if err != nil {
+		return err
+	}
+	// Print the Pod's ID
+	fmt.Println(pod.ID())
+
+	podInfraID, err := pod.InfraContainerID()
+	if err != nil {
+		return err
+	}
+
+	namespaces = map[string]string{
+		"pid":  fmt.Sprintf("container:%s", podInfraID),
+		"net":  fmt.Sprintf("container:%s", podInfraID),
+		"user": fmt.Sprintf("container:%s", podInfraID),
+		"ipc":  fmt.Sprintf("container:%s", podInfraID),
+		"uts":  fmt.Sprintf("container:%s", podInfraID),
+	}
+	if !c.Bool("quiet") {
+		writer = os.Stderr
+	}
+
+	dockerRegistryOptions := image2.DockerRegistryOptions{
+		DockerRegistryCreds: registryCreds,
+		DockerCertPath:      c.String("cert-dir"),
+	}
+	if c.IsSet("tls-verify") {
+		dockerRegistryOptions.DockerInsecureSkipTLSVerify = types.NewOptionalBool(!c.BoolT("tls-verify"))
+	}
+
+	for _, container := range podYAML.Spec.Containers {
+		var (
+			containerConfig createconfig.CreateConfig
+			envs            map[string]string
+		)
+
+		containerConfig.Runtime = runtime
+		containerConfig.Image = container.Image
+		newImage, err := runtime.ImageRuntime().New(ctx, container.Image, c.String("signature-policy"), c.String("authfile"), writer, &dockerRegistryOptions, image2.SigningOptions{}, false)
+		if err != nil {
+			return err
+		}
+		containerConfig.ImageID = newImage.ID()
+		containerConfig.Name = container.Name
+		containerConfig.Tty = container.TTY
+		containerConfig.WorkDir = container.WorkingDir
+		containerConfig.ReadOnlyRootfs = *container.SecurityContext.ReadOnlyRootFilesystem
+		containerConfig.Privileged = *container.SecurityContext.Privileged
+		// TODO find the home for allowPrivilegeEscalation in containerConfig
+
+		containerConfig.Command = container.Command
+		containerConfig.StopSignal = 15
+
+		// If the user does not pass in ID mappings, just set to basics
+		if containerConfig.IDMappings == nil {
+			containerConfig.IDMappings = &storage.IDMappingOptions{}
+		}
+
+		containerConfig.NetMode = ns.NetworkMode(namespaces["net"])
+		containerConfig.IpcMode = ns.IpcMode(namespaces["ipc"])
+		containerConfig.UtsMode = ns.UTSMode(namespaces["uts"])
+		containerConfig.PidMode = ns.PidMode(namespaces["pid"])
+		containerConfig.UsernsMode = ns.UsernsMode(namespaces["user"])
+
+		if len(container.Env) > 0 {
+			envs = make(map[string]string)
+		}
+		// Environment Variables
+		for _, e := range container.Env {
+			envs[e.Name] = e.Value
+		}
+		containerConfig.Env = envs
+
+		runtimeSpec, err := createconfig.CreateConfigToOCISpec(&containerConfig)
```

Comment:
Can we share this with podman create/run? It seems like basically the exact same sequence of operations we use there